tests/expose_test.go

```diff
@@ -0,0 +1,384 @@
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/expose"
+	"kubevirt.io/kubevirt/pkg/virtctl/offlinevm"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Expose", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+	const testPort = 1500
+
+	Context("Expose service on a VM", func() {
+		By("Creating a VM object")
+		vm := tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+		vm.Labels = map[string]string{"expose": "vm"}
+
+		It("Should start a VM and run a 'hello world' server on it", func() {
+			By("Starting the VM")
+			vm, err = virtClient.VM(tests.NamespaceTestDefault).Create(vm)
```

Comment:
Hm, that only works by accident. You can't assume a specific execution order for `It`. You will have to use the same pattern like in the networking tests, with the `BeforeAll` and `AfterAll`. Basically make a `BeforeAll` out of this.

Reply:
ok. will fix